smart_open/s3.py

```diff
@@ -6,6 +6,7 @@
 # from the MIT License (MIT).
 #
 """Implements file-like objects for reading and writing from/to AWS S3."""
+from __future__ import annotations
```

Comment:
@mpenkov  this breaks compatibility for py3.6

Reply:
we no longer support Py3.6